tests/components/envisalink/test_sensor.py

```diff
@@ -0,0 +1,21 @@
+"""Tests for envisalink sensor."""
+
+import pytest
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+from homeassistant.util import slugify
+
+from .common import PARTITIONS, setup_platform
+
+
+@pytest.mark.parametrize("expected_lingering_tasks", [True])
```

Comment:
Can you please explain why this is needed? I spent a long time adding this check and it highlights an underlying bug IMO

Reply:
Setting up the envisalink platform sets up a watchdog task and a task that retries to connect on failure.
When shutting down these tasks are still running and cause a pytest failure.
I see many other tests use this same strategy.

Reply:
The watchdog tasks should be cancelled on shutdown.
This is a bug that needs addressing in a separate PR.

I suggest that you use a preliminary PR for tests+task cancellation, then revisit this one.

Reply:
Also see #88905